core/break.lua

```diff
@@ -74,34 +75,60 @@ function lineBreak:trimGlue() -- 842
   nodes[#nodes+1] = SILE.nodefactory.penalty(inf_bad)
 end
 
+function lineBreak:parShape()
+  return { width = self.hsize, left = 0, right = 0 }
```

Comment:
I would like to refactor this a little bit to make the end-user interface more flexible. Rather than expecting a table with mixed types and line data that is clearly redundant mathematically, I would like this function to return 3 values, any one of which can be nil to use defaults, and all of which may be measurements.

Reply:
Maybe then, for consistency, should we also make the hangIndent a measurement rather than an integer, what do you think about it?

Reply:
Yes, it should be.

Reply:
I'm not sure about this one. There is a bit of difference between lua function arguments and a declared setting that has a type that we actually typecheck. Given that the typesetter only actually *utilizes* an integer here, accepting the setting as `integer or nil` seems like the most logical and lowest friction thing to do. Anything else introduces a weird edge case where it isn't clear to the end user when the setting would be `:absolute()`ised. At time of setting? At paragraph init? At time of use? Once and done? An every paragraph?

I am struggling to see the utility in that.

I setup the `lineBreak:parShape()` function return values to accept anything and cast them to what we use internally. Since the entire mechanism is cached this is pretty low overhead. This doesn't seem equivalent of the hangIndent setting.

Reply:
> I am struggling to see the utility in that.

For API consistency, and the setting to accept eg. "2cm" rather than an unqualified integer value (i.e. absolute points).

Reply:
Interestingly the way we have these parameters setup all SILE settings are automatically cast to their expected types at assignment time. This means you can already pass "2cm" just fine. The difference in changing the setting from a number to a measurement will change when relative values (i.e. not "2cm" but "2%fw") are absolutized: at assignment on shortly prior to use. The later sounds potentially more useful to me so I'll go ahead and make the handIndent setting measurement.